.github/workflows/ci.yml

```
@@ -102,6 +114,18 @@ jobs:
           flags: ${{ matrix.test-type }}
           env_vars: OS,PYTHON
 
+      - name: Upload test results
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          # name has to be unique, to not overwrite uploads of other matrix runs. sha1 is optional and only to differentiate
+          # when locally dowloading and comparing results of different workflow runs.
+          name: test-results-${{ github.sha }}-linux-${{ matrix.python-version }}-${{ matrix.test-type }}-${{ matrix.test-group }}
```

```suggestion
          name: test-results-${{ github.sha }}-${{ env.OS }}-${{ matrix.python-version }}-${{ matrix.test-type }}-${{ matrix.test-group }}
```
Do you think we could simplify this like this?

---

(would also need updates in the other jobs)

---

The corresponding context property is ${{ runner.os }}. Using that now